applications/asset_tracker/src/main.c

```diff
@@ -502,27 +503,36 @@ static void modem_rsrp_handler(char rsrp_value)
 		return;
 	}
 
-	k_work_submit_to_queue(&application_work_q, &rsrp_work);
+	/* Only send the RSRP if transmission is not already scheduled.
+	 * Checking CONFIG_HOLD_TIME_RSRP gives the compiler a shortcut.
+	 */
+	if (CONFIG_HOLD_TIME_RSRP == 0 ||
+	    k_delayed_work_remaining_get(&rsrp_work) == 0) {
+		k_delayed_work_submit_to_queue(&application_work_q, &rsrp_work,
+					       K_NO_WAIT);
+	}
 }
 
 /**@brief Publish RSRP data to the cloud. */
 static void modem_rsrp_data_send(struct k_work *work)
 {
 	char buf[CONFIG_MODEM_INFO_BUFFER_SIZE] = {0};
-	static u32_t timestamp_prev;
+	static s32_t rsrp_prev; /* RSRP value last sent to cloud */
```

Comment:
Thanks for noticing, the line had slipped the commit. Now on 547. Placed it after the actual transmission, just in case more early returns or similar are added in the future.